src/core.js

```diff
@@ -46,6 +44,8 @@ register([
     require('./components/errorbars'),
     require('./components/colorscale'),
     require('./components/colorbar'),
+    require('./components/legend'),
+    require('./components/fx'), // fx needs to come after legend
```

Comment:
Similar to the `// fx needs to come after legend` comment, maybe add a comment to `legend` about why you moved it after `colorbar`? Otherwise there's no indication that this would matter.

Reply:
Good call. Addressed in [f0c6b72](https://github.com/plotly/plotly.js/pull/6653/commits/f0c6b72b6d9649d695cfaafd6a4548418beeefed).